static/app/views/dashboardsV2/widgetCard/widgetQueries.tsx

```diff
@@ -354,15 +362,9 @@ class WidgetQueries extends Component<Props, State> {
         isMetricsData = isMetricsData === false ? false : data.meta?.isMetricsData;
 
         // Cast so we can add the title.
-        let tableData = data as TableDataWithTitle;
-        // events api uses a different response format so we need to construct tableData differently
-        if (shouldUseEvents) {
-          const fieldsMeta = (data as EventsTableData).meta?.fields;
-          tableData = {
-            ...data,
-            meta: {...fieldsMeta, isMetricsData: data.meta?.isMetricsData},
-          } as TableDataWithTitle;
-        }
+        const tableData = this.config.transformTable!(data, widget.queries[0], {
```

Comment:
I think right now we define `transformTable` on all of our configs. Should we just make it required for now? The idea that we'll need `!` to assert that it's defined makes me a little worried because if a new dataset was added and these were reusable, we could actually get a runtime error since we're telling TS to ignore that it's potentially undefined.

Reply:
oh i actually wanted to clean this up, i'll leave it optional but deal with it with some optional chaining

Reply:
meh, actually don't want to implicitly handle this, so made it required for now